### PR TITLE
EN-1846 provider agnostic core

### PR DESCRIPTION
**Pull request by Will-NOQ**
## What's changed in the PR?
* Removes AWS provider refs from core

## How to test?
* Running existing unit tests and functional tests. The changes are so fundamental that there would've been a lot of failures in the AWS functional tests and unit tests if this didn't work.

## Jira Ticket
https://noqdev.atlassian.net/browse/EN-1846

### CODE REVIEW

**Review by castrapel — DISMISSED** [review]
LGTM, how should we merge, given the version bump?

**Review by smoy — DISMISSED** [review]
nothing blocking, the comments are questions to follow up

File: iambic/plugins/v0_1_0/okta/app/models.py [hunk 1]
```diff
@@ -189,10 +189,9 @@ async def _apply_to_account(
         app_exists = bool(current_app)
         tasks = []
 
-        # Marking for deletion. This shouldn't be done on the fly.
-        # self = await remove_expired_resources(
-        #     self, self.resource_type, self.resource_id
-        # )
+        self = await remove_expired_resources(
```
[the remark this answers is not in the document]
Comment (Will-NOQ) [reply iambic/plugins/v0_1_0/okta/app/models.py]: Yeah I'm not sure how it got added back. I re-removed it. Good catch!

File: iambic/core/template_generation.py [hunk 2]
```diff
@@ -42,22 +41,22 @@ async def get_existing_template_map(
...
-        sanitize_string(aws_account.account_name, valid_characters_re),
-        "{{account_name}}",
-    )
-    for var in aws_account.variables:
-        resource = resource.replace(var.value, "{{{}}}".format(var.key))
+
+    if variables := getattr(provider_child, "variables", None):
+        for var in variables:
+            if isinstance(var.value, str):
+                var.value = sanitize_string(var.value, valid_characters_re)
```
[the remark this answers is not in the document]
Comment (Will-NOQ) [reply iambic/core/template_generation.py:57]: Made valid_characters_re a param per our convo.

**Review by smoy — APPROVED** [review]

**Pull request merged**